examples/experimental/imagenet_gaea/gcp_bucket_helper.py

```diff
@@ -0,0 +1,45 @@
+import os
+import time
+import random
+
+# Helper functions for GCP from https://cloud.google.com/storage/docs/listing-objects#code-samples
+def list_blobs(storage_client, bucket_name, prefix=None):
+    """Lists all the blobs in the bucket."""
+    blobs = storage_client.list_blobs(bucket_name, prefix=prefix)
+    paths = []
+    for b in blobs:
+        paths.append(b.name)
+    return paths
+
+
+def download_blob(storage_client, bucket_name, source_blob_name, destination_file_name, n_retries=4, max_backoff=8):
+    """Downloads a blob from the bucket."""
+
+    bucket = storage_client.bucket(bucket_name)
+    destination_dir = destination_file_name[0:destination_file_name.rfind('/')]
+    if not os.path.exists(destination_dir):
+        try:
+            os.makedirs(destination_dir)
+        except:
+            assert os.exists(destination_dir)
+            pass
+    blob = bucket.blob(source_blob_name)
+    try:
+        for n in range(n_retries):
+            try:
+                return blob.download_to_filename(destination_file_name)
+            except:
+                print("Retrying download...")
+                time.sleep(min(2**n + random.random(), max_backoff))
+    except IOError as e:
+        print(e)
+        raise
+
+
+#if __name__=="__main__":
+    bucket_name = 'determined-ai-datasets'
+    prefix = 'imagenet/imagenet/validation'
+    storage_client = storage.Client()
+    img_paths = list_blobs(storage_client, bucket_name, prefix)
+    print(img_paths[0:10])
+    download_blob(storage_client, bucket_name, img_paths[2], os.path.join('/tmp',img_paths[2]))
```

Comment:
I'm a bit confused about this chunk? Is this meant to be commented out?

Reply:
This was used for my own testing purposes.  I now use the function aaron pointed to for downloading from GCP so this file has been removed.